frontend/src/composables/useMaxFillableAmount.ts

```diff
@@ -0,0 +1,43 @@
+import type { Ref } from 'vue';
+import { computed, ref, shallowRef, watch } from 'vue';
+
+import { type IEthereumProvider, createEthereumProvider } from '@/services/web3-provider';
+import type { Chain, Token } from '@/types/data';
+import type { TokenAmount } from '@/types/token-amount';
+import { AGENT_ADDRESSES } from '@/utils/agentAddresses';
+
+import { useTokenBalance } from './useTokenBalance';
+
+export function useMaxFillableAmount(
+  targetChain: Ref<Chain | undefined>,
+  token: Ref<Token | undefined>,
+) {
+  const provider: Ref<IEthereumProvider | undefined> = shallowRef(undefined);
+  watch(
+    targetChain,
+    async (chain) =>
+      (provider.value = chain ? await createEthereumProvider(chain.internalRpcUrl) : undefined),
+  );
+
+  const balances: Ref<TokenAmount | undefined>[] = [];
+  for (const agent of AGENT_ADDRESSES) {
+    const { balance } = useTokenBalance(provider, ref(agent), token);
```

Comment:
Can you explain how this work? To me it is very confusing. The second argument to useTokenBalance is a token, here we are passing a ref to the agent?